### PR TITLE
MSM FFA: Create msm/ffa folder & structure

### PR DESCRIPTION
**Pull request by volhovm**
Creates a directory structure for having `/ffa` code. 

Parent branch PR: https://github.com/o1-labs/proof-systems/pull/1839

Part of #1790.

### CODE REVIEW

File: msm/src/main.rs [hunk 1]
```diff
@@ -41,7 +42,7 @@ pub fn main() {
 
     let circuit_env = generate_random_msm_witness();
     let proof_inputs = circuit_env.get_witness();
-    let constraint_exprs = circuit_env.get_exprs_add();
```
Comment (dannywillems) [inline msm/src/main.rs:44]: For a follow-up: we can move the main file in the subdirectory I think

File: msm/src/expr.rs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+///     ExprInner::<Operations<ConstantExprInner<Fp>>, Column>::Cell(Variable {
+///         col: Column::X(1),
+///         row: CurrOrNext::Curr,
+///     }),
+/// );
+/// let constraint = x3 - x1 - x2;
+/// ```
+/// A list of such constraints is used to represent the entire circuit and will
+/// be used to build the quotient polynomial.
+pub type MSMExpr<F> = Expr<ConstantExpr<F>, Column>;
```
Comment (dannywillems) [inline msm/src/expr.rs:45]: For a follow-up: I think `expr.rs` can be deleted. We only need the generalized definition of  `Column`.

**Review by dannywillems — APPROVED** [review]
LGTM. Need to merge master into the branch.

File: msm/src/main.rs (hunk 1 above)
Comment (volhovm) [reply msm/src/main.rs:44]: Will do

**Pull request merged**